openwisp_monitoring/device/admin.py

```diff
@@ -63,10 +63,32 @@ class CheckInline(GenericStackedInline):
     fields = ['check_type', 'is_active']
     readonly_fields = ['check_type']
 
+    def _set_check_inline(self):
+        self.fields = []
+        self.readonly_fields = ['created', 'modified']
+
     def has_add_permission(self, request, obj=None):
+        if request.user.has_perm(f'{self.model._meta.app_label}.add_check_inline'):
+            self._set_check_inline()
+            return True
+        return False
+
+    def has_view_permission(self, request, obj=None):
+        if request.user.has_perm(f'{self.model._meta.app_label}.view_check_inline'):
+            self._set_check_inline()
+            return True
+        return super(GenericStackedInline, self).has_view_permission(request, obj)
```

Comment:
Why did you only extend the `has_view_permission` method but override other method?

Reply:
I change this way to make sure `Inlines` works similar to our current version of code, i.e. When we have model permissions other than `view`, then they should **add/change/delete** only through  `CheckAdmin` & `MetricAdmin` but not from device `Inlines`.

**Ref** : https://github.com/openwisp/openwisp-monitoring/blob/52cf287a31026874fe73948badbbe174adfd250d/openwisp_monitoring/device/admin.py#L66-L70

https://github.com/openwisp/openwisp-monitoring/blob/52cf287a31026874fe73948badbbe174adfd250d/openwisp_monitoring/device/admin.py#L95-L99

https://github.com/openwisp/openwisp-monitoring/blob/52cf287a31026874fe73948badbbe174adfd250d/openwisp_monitoring/device/admin.py#L112-L116